src/components/TransactionStatus.tsx

```diff
@@ -101,29 +120,40 @@ function TransactionStatus({ transaction, onComplete }: Props) {
       return;
     }
 
-    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null) => {
+    const stepEvaluator = (transactionValue: string | number | null, chainValue: string | number | null): boolean => {
       if (!transactionValue || !chainValue) return false;
 
       const bnChainValue = new BN(chainValue);
       const bnTransactionValue = new BN(transactionValue);
       return bnChainValue.gt(bnTransactionValue);
     };
 
-    const completionStatus = (status: boolean) =>
-      transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
+    const completionStatus = (status: boolean = false): TransactionStatusEnum => {
+      if (transaction.id === 0) {
+        return TransactionStatusEnum.NOT_STARTED;
+      } else if (!status) {
+        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
+      } else {
+        return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

Comment:
```suggestion
      }
      
      if (!status) {
        return transaction.block ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
      }
      
      return transaction.block && status ? TransactionStatusEnum.COMPLETED : TransactionStatusEnum.IN_PROGRESS;
```

I find the early-exit pattern much easier to follow when the rest is not nested in `else` branch, cause the conditions are in visually-predictable locations.